apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -1446,21 +1447,49 @@ def create_network_postcommit(self, context):
             vrf = self._get_network_vrf(context, context.current)
 
             # Create BD and EPG for this network
+            app_profile_name = self._get_network_app_profile(context.current)
             with self.apic_manager.apic.transaction() as trs:
                 self.apic_manager.ensure_bd_created_on_apic(
                     tenant_id, bd_name, ctx_owner=vrf['aci_tenant'],
                     ctx_name=vrf['aci_name'], transaction=trs)
                 self.apic_manager.ensure_epg_created(
                     tenant_id, epg_name,
-                    app_profile_name=self._get_network_app_profile(
-                        context.current), bd_name=bd_name,
+                    app_profile_name=app_profile_name, bd_name=bd_name,
                     transaction=trs)
+            self.apic_manager.update_name_alias(
+                self.apic_manager.apic.fvBD, tenant_id, bd_name,
+                nameAlias=network_name)
+            self.apic_manager.update_name_alias(
+                self.apic_manager.apic.fvAEPg, tenant_id, app_profile_name,
+                epg_name, nameAlias=network_name)
         elif self.fabric_l3:
             self._create_real_external_network(context, context.current)
 
     def update_network_postcommit(self, context):
         super(APICMechanismDriver, self).update_network_postcommit(context)
 
+        # The following validation is not happening in the precommit to avoid
+        # database lock timeout
+        if context.current['name'] == acst.APIC_SYNC_NETWORK:
```

Comment:
you shouldn't be able to create a network with such name in a first place, since it will be deleted once the sync is completed, why is this check needed?